File: src/jvmMain/java/io/github/landgrafhomyak/chatwars/ny2023_map/Main.java
package io.github.landgrafhomyak.chatwars.ny2023_map;

import io.github.landgrafhomyak.chatwars.ny2023_map.db.DatabaseException;
import io.github.landgrafhomyak.chatwars.ny2023_map.db.SqliteDatabase;
import org.sqlite.jdbc4.JDBC4Connection;

import java.io.IOException;
import java.sql.SQLException;
import java.util.Properties;

public final class Main {
    public static void main(String[] argv) throws SQLException, IOException, DatabaseException {
        final String dbpath = System.getProperty("sqlite");
        final SiteRunner server = new SiteRunner(
                new SqliteDatabase(new JDBC4Connection(String.format("file:%s", dbpath), dbpath, new Properties()))
        );
        final Thread botThread = new Thread(
                new BotRunner(
                        System.getProperty("token"),
                        server
                )
        );
        botThread.setDaemon(false);
        botThread.start();
    }

}

File: src/jvmMain/kotlin/io/github/landgrafhomyak/chatwars/ny2023_map/Main.kt
package io.github.landgrafhomyak.chatwars.ny2023_map

import io.github.landgrafhomyak.chatwars.ny2023_map.bot.BotRunner
import io.github.landgrafhomyak.chatwars.ny2023_map.db.SqliteDatabase
import io.github.landgrafhomyak.chatwars.ny2023_map.server.SiteServer
import org.sqlite.jdbc4.JDBC4Connection
import java.util.*


public object Main {
    @JvmStatic
    private fun Runnable.runInThread(): Thread {
        val t = Thread(this)
        t.isDaemon = false
        t.start()
        return t
    }

    @Suppress("UNUSED_VARIABLE")
    @JvmStatic
    public fun main(argv: Array<String>) {
        val dbpath = System.getProperty("sqlite")

        @Suppress("UNUSED_VARIABLE")
        val server = SiteServer(
            SqliteDatabase(JDBC4Connection(String.format("file:%s", dbpath), dbpath, Properties()))
        )
        val syncThread = SyncWithOpponentsRunner(server.database()).runInThread()

        val botThread = BotRunner(System.getProperty("token"), server.database()).runInThread()
    }
}